src/coreservices.cpp

```diff
@@ -136,6 +132,25 @@ void CoreServices::initialize(QApplication* pApp) {
             m_cmdlineArgs.getLogLevel(),
             m_cmdlineArgs.getLogFlushLevel(),
             logFlags);
+}
+
+void CoreServices::preInitialize(QApplication* pApp) {
```

Comment:
Is this method really necessary? Couldn't we just make it part of the CoreServices constructor?

Reply:
Is there any need to create the `CoreServices` object before creating a `QApplication`? If not, we should move this into the constructor. Less possibilities to use this class wrong or miss an initialization step.

Reply:
Ping

Reply:
Done.

Reply:
There does not seem to be a purpose for this method. Move the code directly into the constructor.

Reply:
It is just documentation support.

Reply:
I don't think this makes the code more readable; it makes me wonder why the function exists. I think it would be better to move the code into the constructor and add a comment: "These need to be initialized before the GUI. They are fast to initialize. Everything else is deferred to the `initialize` method so the GUI can show the progress."

Reply:
I like the current state.

Reply:
I don't to bike-shed, but
- it only contains the body of the constructor, which now contains just this single invocation and
- the method is public although it doesn't need to be.

Reply:
The function is private.